inorbit_edge/robot.py

```diff
@@ -899,72 +902,106 @@ def publish_odometry(
         msg.speed_available = True
         self.publish_protobuf(MQTT_SUBTOPIC_ODOMETRY, msg)
 
-    def publish_laser(
-        self, x, y, yaw, ranges, angle=(-math.pi, math.pi), frame_id="map", ts=None
-    ):
-        """Publish robot laser scan
+    def publish_lasers(self, x, y, yaw, ranges, angles, frame_id="map", ts=None):
+        """Publish an array of lasers.
 
         Args:
             x (float): Robot pose x coordinate.
             y (float): Robot pose y coordinate.
             yaw (float): Robot yaw (radians).
-            ranges (List[float]): Laser scan range data. This list of ``float`` number
-                may contain infinite values represented as ``math.pi``.
-            angle (tuple, optional): Laser scan range angle (radians). This parameter
+            ranges (List[List[float]]): A list of Laser scan range data. This list of
+                ``float`` number may contain infinite values represented as ``math.pi``.
+            angles (List[tuple]): Laser scan range angle (radians). This parameter
                 defines the cone in which the laser points will be shown. For full 360
-                degrees scans use (-math.pi, math.pi). Defaults to (-math.pi, math.pi).
+                degrees scans use (-math.pi, math.pi). This list must match the length
+                of ranges.
             frame_id (str, optional): Robot map frame identifier. Defaults to "map".
             ts (int, optional): Pose timestamp. Defaults to int(time() * 1000).
         """
-
-        if not self._should_publish_message(method="publish_laser"):
+        if len(ranges) != len(angles):
+            raise ValueError("Input lists must be of the same length")
```

Comment:
I would prefer to ignore the message and log an error instead of raising an exception. I think it makes sense to do so only on methods invoked when initializing the robot class, so we are resilient to malformed messages